packages/typespec-ts/src/modular/buildOperationUtils.ts

```diff
@@ -0,0 +1,871 @@
+import { toPascalCase } from "../utils/casingUtils.js";
+import { importSettings } from "../utils/importUtils.js";
+import {
+  getResponseMapping,
+  getRequestModelMapping,
+  serializeRequestValue,
+  deserializeResponseValue
+} from "./helpers/operationHelpers.js";
+import { ModularCodeModel, Property, Type } from "./modularCodeModel.js";
+import {
+  FunctionDeclarationStructure,
+  SourceFile,
+  StructureKind
+} from "ts-morph";
+import { Imports as RuntimeImports } from "@azure-tools/rlc-common";
+import { UsageFlags } from "@typespec/compiler";
+
+/**
+ * This function creates serialize and deserialize utils for special unions and that are used in the operation.
+ */
+export function buildOperationUtils(model: ModularCodeModel) {
+  const serializeUtilFiles = [];
+  for (const serializeType of ["serialize", "deserialize"]) {
+    const usageCondition =
+      serializeType === "serialize" ? UsageFlags.Input : UsageFlags.Output;
+    const specialUnions = model.types.filter(
+      (t) =>
+        !isSpecialExcludeUnion(t) &&
```

Comment:
Not sure we could enum all excludes cases, if not is it possible to set the white list explicitly? Any items not in that list would be taken as not supported.